packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// no-unused-vars is temporarily disabled until more functions are implemented
+import { CAIP2, caip2, caip19 } from '@shapeshiftoss/caip'
+import { AssetNamespace } from '@shapeshiftoss/caip/dist/caip19/caip19'
+import { CosmosSignTx } from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+import { cosmos } from '@shapeshiftoss/unchained-client'
+import WAValidator from 'multicoin-address-validator'
+
+import { ChainAdapter as IChainAdapter } from '../api'
+import { ChainAdapter } from '../bitcoin'
+import { ErrorHandler } from '../error/ErrorHandler'
+
+export type CosmosChainTypes = ChainTypes.Cosmos | ChainTypes.Osmosis
+
+export interface ChainAdapterArgs {
+  chainId?: CAIP2
+  providers: {
+    http: cosmos.api.V1Api
+    // unchained-client 5.1.1 does not have a websocket client for cosmos
+  }
+  symbol: string
+}
+
+export abstract class CosmosSdkBaseAdapter<T extends CosmosChainTypes> implements IChainAdapter<T> {
+  protected readonly chainId: CAIP2
+  protected readonly supportedChainIds: CAIP2[]
+  protected readonly symbol: string
+  protected readonly providers: {
+    http: cosmos.api.V1Api
+  }
+
+  public static readonly defaultBIP44Params: BIP44Params = {
+    purpose: 44,
+    coinType: 118,
+    accountNumber: 0
+  }
+
+  protected constructor(args: ChainAdapterArgs) {
+    if (args.chainId && this.supportedChainIds.includes(args.chainId)) {
+      this.chainId = args.chainId
```

Comment:
Nice. I think this is the way to go, having a base Cosmos SDK adapter that can be extended with `chainId` specifically for cosmos or osmosis.